client/ayon_hiero/api/otio/utils.py

```diff
@@ -78,3 +80,37 @@ def get_rate(item):
         return rate
 
     return round(rate, 4)
+
+
+def get_marker_from_clip_index(otio_timeline, clip_index):
+    """
+    Args:
+        otio_timeline (otio.Timeline): The otio timeline to inspect
+        clip_index (int): The clip index metadata to retrieve.
+
+    Returns:
+        tuple(otio.Clip, otio.Marker): The associated clip and marker
+            or (None, None)
+    """
+    try:  # opentimelineio >= 0.16.0
+        all_clips = otio_timeline.find_clips()
+    except AttributeError:  # legacy
+        all_clips = otio_timeline.each_clip()
+
+    # Retrieve otioClip from parent context otioTimeline
+    # See collect_current_project
+    for otio_clip in all_clips:
+        for marker in otio_clip.markers:
+
+            try:
+                json_metadata = marker.metadata["json_metadata"]
+                metadata = json.loads(json_metadata)
```

Comment:
To make it clearer where you're catching the KeyError on you could move the `json.loads()` call also to the `else` statement?

Reply:
definitely, this is done !